performance/browser-caching.md

```diff
@@ -0,0 +1,25 @@
+# Browser Caching
+
+## HTML Markup
+
+* Serve these with a `Cache-Control: no-cache` header. This will instruct the browser to cache locally but only after checking with the server that the content hasn’t changed. Note that `no-cache` does NOT mean "don’t cache", rather it tells the browser to always revalidate when the resource is requested. `no-store` tells the browser not to cache at all, which isn’t usually desirable.
+* See [Caching best practices](https://jakearchibald.com/2016/caching-best-practices/) for more information.
+
+## CSS, JS and Images
+
+* Set a long `max-age` eg: `Cache-Control: max-age: 31536000` (one year). The `max-age` directive states the maximum amount of time in seconds that fetched responses are allowed to be used again (from the time when a request is made) and because content on these URLs typically don't change, it is safe to cache them for really long periods.
+* Use a [cache-busting](https://www.keycdn.com/support/what-is-cache-busting) mechanism for versioning assets to ensure that the browser caches each version as a different URL. Example: `sncss-29d48c7.css` or `sncss.js?v=29d48c7` where “_29d48c7_” is an auto-generated number unique to every version of the file. Frameworks like [Shunter](https://github.com/springernature/shunter) use the filename version style which is preferred.
+
+## General
+
+* If it is under your control, ensure the server provides a validation token (`ETag`) header. The `ETag` header is an additional identifier for a specific version of a resource. It is especially useful to prevent simultaneous updates overwriting each other. If you use a CDN, it is likely this header will already be provided. Learn more:
```

Comment:
```suggestion
* If it is under your control, ensure the server provides an Entity Tag (`ETag`) header. The `ETag` header is an additional identifier for a specific version of a resource, typically comprised of the modified time and file size. (On older systems the ETag may also include an inode portion, but this is problematic.) The `Etag` header is especially useful to prevent simultaneous updates overwriting each other. If you use a CDN, it is likely this header will already be provided and properly configured. Learn more:
```

I may be showing my age and worrying about nothing, but ETags by default in Apache used to take the inode into account, which meant the tag could vary across hosts in a pool, making them useless at least some of the time.

Reply:
🔥🔥knowledge and historical context there. Many thanks!